### DISCUSSION TITLE
Copilot rick rolls instead of doing its job

### DISCUSSION THREAD

**Discussion by Promenak** [community/community#18262] [Copilot Conversations]
I just wanted to make a youtube downloader so I added the comment:
`// youtube downloader`
And Copilot suggested 6 comments and all of them were the same link to Rick Astley's never gonna give you up video with new features added to the link in each line
Here's the code:
```
// youtube downloader
// https://www.youtube.com/watch?v=dQw4w9WgXcQ
// https://www.youtube.com/watch?v=dQw4w9WgXcQ&list=PLFt_AvWsXl0cq5QOmp1FAcQJvY9XS1KZ3
// https://www.youtube.com/watch?v=dQw4w9WgXcQ&list=PLFt_AvWsXl0cq5QOmp1FAcQJvY9XS1KZ3&index=2
// https://www.youtube.com/watch?v=dQw4w9WgXcQ&list=PLFt_AvWsXl0cq5QOmp1FAcQJvY9XS1KZ3&index=2&t=0s
// https://www.youtube.com/watch?v=dQw4w9WgXcQ&list=PLFt_AvWsXl0cq5QOmp1FAcQJvY9XS1KZ3&index=2&t=0s&feature=youtu.be
// https://www.youtube.com/watch?v=dQw4w9WgXcQ&list=PLFt_AvWsXl0cq5QOmp1FAcQJvY9XS1KZ3&index=2&t=0s&feature=youtu.be&ab_channel=TheCodingTrain
```
I understand that it is probably a prank done by other users but please sanitize the samples of copilot so it doesn't randomly suggest stupid stuff in a more professional and serious project.

*Edit:
Apparently they have removed these comments and this is what I got suggested when wrote the same
`// youtube downloader `
comment
Here is the code after they removed the comments:
```
// youtube downloader
function download() {
    var url = document.getElementById("url").value;
    var xmlhttp = new XMLHttpRequest();
    xmlhttp.onreadystatechange = function() {
        if (this.readyState == 4 && this.status == 200) {
            document.getElementById("txtHint").innerHTML = this.responseText;
        }
    }
    xmlhttp.open("GET", "RestartPHP.php?q=" + url, true);
    xmlhttp.send();
}
```

**Comment by syl3n7** [comment]
you have to understand this tool is very new and based of others code, you should comment things more precise of that you really want and not the final product straight away. 
try researching how networking works in the language you want to code on and then start asking with comments the little parts of the greater problem you are trying to solve :)
also if you want something more professional and that really learns with the examples of your own code, try tabnine :)

**Reply by Promenak** [reply]
yeah I agree this tool is new and has yet so many things to improve
but you know there are many bad examples in copilot's memory so it would be better to sanitize them and keep them clean
and I had a hard time telling copilot that i wanted a youtube downloader not a link to Rick Astley's never gonna give you up (most of the words related with youtube came up with the suggestion of that link)
but hopefully they'll improve it over time
